src/Plaid/Link/OnEventMetadata.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Text.Json;
+using Going.Plaid.Entity;
+using Going.Plaid.Exceptions;
+using Newtonsoft.Json;
+
+namespace Going.Plaid.Link
+{
+	/// <summary>
+	/// </summary>
+	/// <remarks>
+	/// See: <see href="https://plaid.com/docs/link/web/#onevent"/><br/>
+	/// <example>
+	/// Reference JSON for building this object:
+	/// <code>
+	/// {
+	///   "error_code": String,
+	///   "error_message": String,
+	///   "error_type": String,
+	///   "exit_status": String,
+	///   "institution_id": String,
+	///   "institution_name": String,
+	///   "institution_search_query": String,
+	///   "link_session_id": String,
+	///   "mfa_type": String,
+	///   "request_id": String,
+	///   "timestamp": String,
+	///   "view_name": String
+	/// }
+	/// </code>
+	/// </example>
+	/// </remarks>
+	public class OnEventMetadata
+	{
+		/// <summary>
+		/// The Event Name is passed to the onEvent Javascript callback as a separate parameter from 
+		/// the metadata structure itself. It must be manually added to the metadata in order to be 
+		/// included when deserializing.<br/>
+		/// <example>
+		/// <code>
+		///  onEvent: function(eventName, metadata)
+		///  {
+		///	  // Optionally capture Link flow events, streamed through
+		///	  // this callback as your users connect an Item to Plaid.
+		///	  // For example:
+		///	  // eventName = "TRANSITION_VIEW"
+		///	  // metadata  = {
+		///	  //   link_session_id: "123-abc",
+		///	  //   mfa_type:        "questions",
+		///	  //   timestamp:       "2017-09-14T14:42:19.350Z",
+		///	  //   view_name:       "MFA",
+		///	  // }
+		///	  console.info('onEvent:' + eventName, metadata);
+		///	  metadata.event_name = eventName;
+		///	  ... send the metadata as JSON to be deserialized into this class ...  
+		///  }
+		/// </code>
+		/// </example>
+		/// </summary>
+		[JsonProperty("event_name")]
+		[Newtonsoft.Json.JsonConverter(typeof(Newtonsoft.Json.Converters.StringEnumConverter))]
```

Comment:
Can we simplify names here? `Newtonsoft.Json` doesn't appear to be necessary, since we have the `using` statement for it. Add one for `Newtonsoft.Json.Converters` as well?

Reply:
oops. my bad